python/ray/data/datasource/delta_sharing_datasource.py

```diff
@@ -0,0 +1,128 @@
+import logging
+from json import loads
+from typing import List, Optional, Tuple
+
+import numpy as np
+from delta_sharing.converter import to_converters
+from delta_sharing.protocol import Table
+from delta_sharing.reader import DeltaSharingReader
+from delta_sharing.rest_client import DataSharingRestClient
```

Comment:
This should be imported in function level. `delta_sharing` should be an optional package, but not a required package. Use `_check_import` instead, example - https://github.com/ray-project/ray/blob/master/python/ray/data/datasource/lance_datasource.py#L29 .